src/main/java/io/dockstore/provision/GSPlugin.java

```diff
@@ -110,6 +114,16 @@ protected String getBlobName(String path) {
             return String.join(File.separator, splitPathListNoBucket);
         }
 
+        private int getBufferSizeToUse() {
+            // Determine how much heap space is available for our buffer
+            long heapAvailable = Runtime.getRuntime().freeMemory();
+            // We will use a percentage of available heap bytes for the buffer
+            long heapBytesToUse = (long)(heapAvailable * PERCENT_OF_HEAP_SPACE_TO_USE);
+            // Limit the size of the buffer to a maximum
+            int bufferSizeToUse =  (int)min(heapBytesToUse, MAX_BUFFER_SIZE);
+            return bufferSizeToUse;
```

Comment:
I think you should have a minimum buffer size, let's say 1Mb.

Looking at [this page](https://stackoverflow.com/questions/3571203/what-are-runtime-getruntime-totalmemory-and-freememory), free memory is not the maximum memory that can be allocated, it's the max memory free right then -- it can be increased the VM allocating more memory and/or garbage collection.

Reply:
added comment